src/services/sessionStorage.ts

```diff
@@ -0,0 +1,129 @@
+import { Either, isLeft, left, right } from "fp-ts/lib/Either";
+import { Op } from "sequelize";
+import { Session } from "../models/Session";
+import { User } from "../models/User";
+import { SpidLoggedUser, SpidUser } from "../types/spidUser";
+import { SessionToken } from "../types/token";
+import { log } from "../utils/logger";
+
+export const sessionNotFoundError = new Error("Session not found");
+
+export default class SessionStorage {
+  public static async set(
+    user: SpidUser,
+    sessionToken: SessionToken
+  ): Promise<Either<Error, boolean>> {
+    const TOKEN_DURATION_IN_SECONDS = Number(
+      process.env.TOKEN_DURATION_IN_SECONDS
+    );
+    if (!TOKEN_DURATION_IN_SECONDS) {
+      log.error("TOKEN_DURATION_IN_SECONDS environment variable is missing");
+      return process.exit(1);
+    }
+
+    try {
+      const [loggerUser, _] = await User.findOrCreate({
+        defaults: {
+          email: user.email,
+          familyName: user.familyName,
+          firstName: user.name
+        },
+        where: { fiscalCode: user.fiscalNumber }
+      });
+      await loggerUser.createSession({
+        expirationTime: new Date(Date.now() + TOKEN_DURATION_IN_SECONDS),
+        token: sessionToken
+      });
+    } catch (error) {
+      return left<Error, boolean>(
+        new Error("Error creating session for the user")
+      );
+    }
+    return right<Error, boolean>(true);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static async getBySessionToken(
+    token: SessionToken
+  ): Promise<Either<Error, User>> {
+    const errorOrSession = await this.loadSessionBySessionToken(token);
+
+    if (isLeft(errorOrSession)) {
```

Comment:
wait, do you understand what the following code does? do you think it's necessary?